### PR TITLE
fix(styles): update border-block-end to use $border-subtle-01

### LINKED ISSUE #17454: [Bug]: Data table border-subtle is off by a layer
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.66.0

### React version

_No response_

### Description

The layer structure for the border-subtles in data table is off by one. It should start at `$border-subtle-01` instead of `$border-subtle-02` Similar problem to what was fixed in https://github.com/carbon-design-system/carbon/pull/13879.

image

### Reproduction/example

n/a

### Steps to reproduce

Open up the data table example in storybook and switch to the Gray 10 theme. The border color that appears on the white table is too dark.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

Core but Cloud brought this to my attention 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by ziyadzulfikar**
Closes #17454

Updated the `border-block-end` of each <td> element in the table rows.

#### Changelog

**Changed**

- Changed `border-block-end: 1px solid $border-subtle;` to `border-block-end: 1px solid $border-subtle-01;`

#### Testing / Reviewing

- Navigate to the DataTable component's playground story.

- Review the styling of the `<td>` elements in the table rows that use the `border-block-end` property, and ensure the values are applied correctly and consistently.

### CODE REVIEW

**Comment by ziyadzulfikar** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/styles/scss/components/data-table/_data-table.scss [hunk 1]
```diff
@@ -205,7 +205,7 @@
 
   .#{$prefix}--data-table td,
   .#{$prefix}--data-table tbody th {
-    border-block-end: 1px solid $border-subtle;
+    border-block-end: 1px solid $border-subtle-01;
```
Comment (tay1orjones) [inline packages/styles/scss/components/data-table/_data-table.scss:208]: Similar to [this PR](https://github.com/carbon-design-system/carbon/pull/13879/files#diff-417d728c7ed2601e1b3cd4667e5c281ea3de98d57769d1919146c42b6ceaf7ac), additional selectors/styles are needed to ensure the token is updated appropriately. `.#{$prefix}--layer-two`, `.#{$prefix}--layer-three`, etc.

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Thanks for this! There is one small piece still needed though, see below

File: packages/styles/scss/components/data-table/_data-table.scss (hunk 1 above)
Comment (tay1orjones) [reply packages/styles/scss/components/data-table/_data-table.scss:208]: Without this update the cells will not respond at all to being placed on another layer and will be "locked" to the layer 01 style. There's a bit more detail about this [on the website](https://carbondesignsystem.com/elements/color/usage/#implementing-layering)

Comment (ziyadzulfikar) [reply packages/styles/scss/components/data-table/_data-table.scss:208]: @tay1orjones I have updated the changes according to your suggestion.

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by guidari — APPROVED** [review]
LGTM! You just have to fix some merge conflicts

**Comment by ziyadzulfikar** [comment]
@guidari 
I have resolved the conflicts.

**Comment by guidari** [comment]
Hey, I'm checking Percy and seems like this changes triggers an error in [Datatable story](https://deploy-preview-17548--v11-carbon-react.netlify.app/?path=/story/components-datatable-withailabel--ai-label-with-selection-and-expansion) with expansion. The line it is not complete

Screenshot 2024-10-14 at 10 31 21

**Comment by ziyadzulfikar** [comment]
@guidari  This issue has been fixed.

> Hey, I'm checking Percy and seems like this changes triggers an error in [Datatable story](https://deploy-preview-17548--v11-carbon-react.netlify.app/?path=/story/components-datatable-withailabel--ai-label-with-selection-and-expansion) with expansion. The line it is not complete
> 
> Screenshot 2024-10-14 at 10 31 21

**Comment by guidari** [comment]
Cool! Percy LGTM!
It is just missing @tay1orjones review now!

**Review by tay1orjones — APPROVED** [review]
LGTM!

**Pull request merged**